docs: update picking sources and unify with patching rules (#14377)
conan-io/conan-center-index: docs/adding_packages/conandata_yml_format.md

In the context of ConanCenterIndex, this file is _mandatory_ and consists of two main sections that will be explained in the
next sections with more detail:

* `sources`: Library sources origin with their verification checksums.

> * `patches`: List of patches the library requires, see the [Patching Policy](sources_and_patches.md#policy-about-patching) for the criteria.

Author: 🤔 I like this but the implication is that the libraries provide these patches which is rarely the case.

I tried to build on it!

```suggestion
* `patches`: A list of patches to apply and supporting information. See the [Patching Policy](sources_and_patches.md#policy-about-patching) for the criteria.
```
